File: solutions/344.Reverse_String/Reverse_String.cpp
# include <iostream>
# include <string>

using namespace std;

class Solution {
public:
    // 雙指針
    string reverseString(string s)
    {
        int i = 0;
        int j = s.size()-1;
        while (i < j)
        {
            char tmp = s[j];
            s[j] = s[i];
            s[i] = tmp;
            i++;
            j--;
        }
        return s;
    }

    /*
    {
        for (int L=0, R=s.size()-1; L<R; L++, R--){
            char tmp = s[R];
            s[R] = s[L];
            s[L] = tmp;
        }
        return s;
    }
    */

};

File: solutions/344.Reverse_String/Reverse_String.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        if len(s) < 2:
            return s
        return self.reverseString(s[len(s)/2:]) + \
            self.reverseString(s[:len(s)/2])

    def reverseStringClassic(self, s):
        # classic shit
        i = 0
        j = len(s) - 1
        s = list(s)
        # for i in xrange(len(s)):
        while i < j:
            tmp = s[i]
            s[i] = s[j]
            s[j] = tmp
            i += 1
            j -= 1
        return ''.join(s)

    def idontknowwtfiwrote(self, s):
        # result = ""
        # length = len(s)
        # for i in xrange(1,length+1):
        #     result += s[length-i]

        # for i in reversed(s):
        #     result += i
        # return result
        pass

    def reverseStringPythonic(self, s):
        return s[::-1]
